### PR TITLE
Update ClusterLocalModel to LocalModelCache

### PR DESCRIPTION
**Pull request by yuzisun**
**What this PR does / why we need it**:
This PR rename the local model custom resource from `ClusterLocalModel` to `LocalModelCache` to align with the vision of adding companion resource `ModelCache` for remote model cache using high speed storage.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Type of changes**
Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**Feature/Issue validation/testing**:

Please describe the tests that you ran to verify your changes and relevant result summary. Provide instructions so it can be reproduced.
Please also list any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

- Logs

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Checklist**:

- [ ] Have you added unit/e2e tests that prove your fix is effective or that this feature works?
- [ ] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

**Release note**:

```release-note

```

**Re-running failed tests**

- `/rerun-all` - rerun all failed workflows.
- `/rerun-workflow <workflow name>` - rerun a specific failed workflow. Only one workflow name can be specified. Multiple /rerun-workflow commands are allowed per comment.

### CODE REVIEW

**Review by terrytangyuan — COMMENTED** [review]
This is just a renaming, right?

**Comment by yuzisun** [comment]
> This is just a renaming, right?

Correct

File: Makefile [hunk 1]
```diff
@@ -87,7 +87,7 @@ deploy-dev: manifests
 	cd config/default && if [ ${KSERVE_ENABLE_SELF_SIGNED_CA} != false ]; then \
 	echo > ../certmanager/certificate.yaml; \
 	else git checkout HEAD -- ../certmanager/certificate.yaml; fi;
-	kubectl apply --server-side=true -k config/overlays/development
+	kubectl apply --server-side=true --force-conflicts -k config/overlays/development
```
Comment (HotsauceLee) [inline Makefile:90]: Do we want `--force-conflicts` on the other 3 `kubectl apply` commands in `deploy-dev` and `deploy` as well?

**Review by HotsauceLee — APPROVED** [review]
lgtm with a small comment

File: Makefile (hunk 1 above)
Comment (yuzisun) [reply Makefile:90]: Need to take a deeper look, when I run locally I had to add this flag to apply the change

File: pkg/apis/serving/v1alpha1/local_model_cache_status.go [hunk 2]
```diff
@@ -43,9 +43,6 @@ const (
 	NodeDownloading     NodeStatus = "NodeDownloading"
 	NodeDownloaded      NodeStatus = "NodeDownloaded"
 	NodeDownloadError   NodeStatus = "NodeDownloadError"
-	NodeDeleting        NodeStatus = "NodeDeleting"
```
Comment (yuzisun) [inline pkg/apis/serving/v1alpha1/local_model_cache_status.go:46]: Remove these as they are not used, @greenmoon55 to confirm

File: go.sum [hunk 3]
```diff
@@ -1,3 +1,4 @@
+cel.dev/expr v0.15.0/go.mod h1:TRSuuV7DlVCE/uwv5QbAiW/v8l5O8C4eEPHeu7gf7Sg=
 cloud.google.com/go v0.26.0/go.mod h1:aQUYkXzVsufM+DwF1aE+0xfcU+56JwCaLick0ClmMTw=
```
Comment (greenmoon55) [inline go.sum]: do a `go mod tidy`?

File: pkg/apis/serving/v1alpha1/local_model_cache_status.go (hunk 2 above)
Comment (greenmoon55) [reply pkg/apis/serving/v1alpha1/local_model_cache_status.go:46]: yeah, they are not used.

**Review by greenmoon55 — COMMENTED** [review]
LGTM

**Review by greenmoon55 — APPROVED** [review]

**Pull request merged**